bin/nyc.js

```diff
@@ -36,21 +34,60 @@ if ([
   }
   if (argv.all) nyc.addAllFiles()
 
-  var env = {
-    // Support running nyc as a user without HOME (e.g. linux 'nobody'),
-    // https://github.com/istanbuljs/nyc/issues/951
-    SPAWN_WRAP_SHIM_ROOT: process.env.SPAWN_WRAP_SHIM_ROOT || process.env.XDG_CACHE_HOME || require('os').homedir(),
-    NYC_CONFIG: JSON.stringify(argv),
-    NYC_CWD: process.cwd()
-  }
+  if (argv.setNodeOptions) {
+    const wrapper = require.resolve('../lib/wrap.js')
+    let nodeRequire
+    /*
+     * Node.js 11 and older do not support specifying `--require "/path/with spaces"`.
+     *  - For node.js 12+ we use the proper quoted syntax.
+     *  - For older versions we just directly use the path to lib/wrap.js as long as
+     *    it does not contain any spaces.
+     *  - For older versions when the path to lib/wrap.js contains a space we must
+     *    combine NODE_PATH and NODE_OPTIONS.
+     *
+     * Ref https://github.com/nodejs/node/pull/24065
+     */
+    /* istanbul ignore next */
+    if (process.versions.node.split('.')[0] >= 12) {
+      nodeRequire = ` --require "${wrapper.replace(/"/g, '\\"')}"`
+    } else if (!wrapper.includes(' ')) {
+      nodeRequire = ` --require ${wrapper}`
+    } else {
+      let nodePath = path.dirname(wrapper)
+      if (process.env.NODE_PATH) {
+        const sep = process.platform === 'win32' ? ';' : ':'
+        nodePath = process.env.NODE_PATH + sep + nodePath
+      }
 
-  if (argv['babel-cache'] === false) {
-    // babel's cache interferes with some configurations, so is
-    // disabled by default. opt in by setting babel-cache=true.
-    env.BABEL_DISABLE_CACHE = process.env.BABEL_DISABLE_CACHE = '1'
-  }
+      process.env.NODE_PATH = nodePath
+      nodeRequire = ` --require wrap.js`
```

Comment:
Using NODE_PATH in this way is probably sketchy.  I'm concerned this could hijack unrelated `require` actions by exposing the `node_modules` in nyc to code that shouldn't be using it.  Likely only an issue if `NODE_PATH` is already being used, but still a concern.